src/azure-cli/azure/cli/command_modules/acs/_loadbalancer.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from distutils.version import StrictVersion  # pylint: disable=no-name-in-module,import-error
+
+# pylint: disable=no-name-in-module,import-error
+from azure.mgmt.containerservice.v2019_11_01.models import ManagedClusterLoadBalancerProfile
+from azure.mgmt.containerservice.v2019_11_01.models import ManagedClusterLoadBalancerProfileManagedOutboundIPs
+from azure.mgmt.containerservice.v2019_11_01.models import ManagedClusterLoadBalancerProfileOutboundIPPrefixes
+from azure.mgmt.containerservice.v2019_11_01.models import ManagedClusterLoadBalancerProfileOutboundIPs
+from azure.mgmt.containerservice.v2019_11_01.models import ResourceReference
+
+
+def set_load_balancer_sku(sku, kubernetes_version):
+    if sku:
+        return sku
+    if kubernetes_version and StrictVersion(kubernetes_version) < StrictVersion("1.13.0"):
+        print('Setting load_balancer_sku to basic as it is not specified and kubernetes \
+        version(%s) less than 1.13.0 only supports basic load balancer SKU\n' % (kubernetes_version))
+        return "basic"
+    return "standard"
+
+
+def update_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
+                                 outbound_ports, idle_timeout, profile):
+    """parse and update an existing load balancer profile"""
+    if not is_load_balancer_profile_provided(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
+                                             outbound_ports, idle_timeout):
+        return profile
+    return configure_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
+                                           outbound_ports, idle_timeout, profile)
+
+
+def create_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
+                                 outbound_ports, idle_timeout):
+    """parse and build load balancer profile"""
+    if not is_load_balancer_profile_provided(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
+                                             outbound_ports, idle_timeout):
+        return None
+
+    profile = ManagedClusterLoadBalancerProfile()
+    return configure_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
+                                           outbound_ports, idle_timeout, profile)
+
+
+def configure_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes, outbound_ports,
+                                    idle_timeout, profile):
+    """configure a load balancer with customer supplied values"""
+    if not profile:
+        return profile
```

Comment:
If profile is null, skip?

Reply:
Thanks for the reminder, appreciate that. 

I don't expect the profile to be null at this point - because the callers of this function only call it when there is a profile. But since I am editing the profile in this func, to avoid a panic/exception I am just double checking that it isn't null.